src/backend/marsha/core/api/video.py

```diff
@@ -1196,3 +1202,37 @@ def jitsi_info(self, request, pk=None):
         )
 
         return Response(jitsi_info)
+
+    @action(methods=["post"], detail=True, url_path="initiate-transcript")
+    # pylint: disable=unused-argument
+    def initiate_transcript(self, request, pk=None):
+        """Initiate the transcript process for the video."""
+        video = self.get_object()
+
+        if video.live_state is not None:
+            return Response(
+                {"detail": "Cannot initiate transcript on a live video"}, status=400
+            )
+
+        stamp = video.get_source_s3_key().split("/")[-1]
+
+        # Launch the PeerTube transcription process
+        if settings.TRANSCODING_CALLBACK_DOMAIN:
+            domain = settings.TRANSCODING_CALLBACK_DOMAIN
+        else:
+            domain = f"{request.scheme}://{request.get_host()}"
+
+        launch_video_transcript.delay(video_pk=video.id, stamp=stamp, domain=domain)
+
+        timed_text_track = TimedTextTrack.objects.create(
+            video=video,
+            language=settings.LANGUAGES[0][0],
+            mode=TimedTextTrack.TRANSCRIPT,
```

Comment:
There is a unicity constraint on that 3 fields. IMO something should be added to handle this unicity. Otherwise we will have errors.